langserver/definition.go

```diff
@@ -87,6 +108,26 @@ func (h *LangHandler) definitionGodef(ctx context.Context, params lsp.TextDocume
 	return fset, res, []lsp.Location{loc}, nil
 }
 
+type foundNode struct {
+	ident	*ast.Ident      // the lookup in Uses[] or Defs[]
+	typ	*types.TypeName // the object for a named type, if present
+}
+
+func (h *LangHandler) typeLookup(prog *loader.Program, typ types.Type) *types.TypeName {
+	if typ == nil {
+		return nil
+	}
+	switch t := typ.(type) {
+	case *types.Pointer:
+		return h.typeLookup(prog, t.Elem())
+	case *types.Slice:
+		return h.typeLookup(prog, t.Elem())
```

Comment:
This looks slightly debatable to me. If I look up something which is `[]foo.Bar`, is it confusing to jump to `foo.Bar`? Obviously `foo.Bar` is the more useful thing to look at though.